### PR TITLE
Added remove nodes stage

### PR DESCRIPTION
**Pull request by leerichard42**

### CODE REVIEW

File: lib/removeUnusedNodes.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    var scenes = gltf.scenes;
+    var nodes = gltf.nodes;
+
+    // Build hash of used nodes by traversing through node trees starting at scenes
+    if (defined(scenes)) {
+        for (var sceneId in scenes) {
+            if (scenes.hasOwnProperty(sceneId)) {
+            	var roots = scenes[sceneId].nodes;
+            	if (defined(roots)) {
+            		var nodeStack = [];
```
Comment (pjcozzi) [inline lib/removeUnusedNodes.js]: Looks like this block is indented 8, instead of 4, spaces.

**Comment by pjcozzi** [comment]
Wow, once again, great job!  Just that minor formatting comment.

Nice work testing the edge cases.

**Comment by pjcozzi** [comment]
Also, I merged in master and fixed the conflicts, so make sure to `git pull`.

**Comment by leerichard42** [comment]
Updated!

**Pull request merged**